test/battle/move_effect/embargo.c

```diff
@@ -0,0 +1,394 @@
+        TURN { MOVE(player, MOVE_EMBARGO); }
+        TURN { MOVE(player, MOVE_SCRATCH); MOVE(opponent, MOVE_SCRATCH); }
+    } SCENE {
+        // Turn 1
+        MESSAGE("Wobbuffet used Embargo!");
+        MESSAGE("Foe Wobbuffet can't use items anymore!");
+        // Turn 2
+        MESSAGE("Foe Wobbuffet used Scratch!");
+        MESSAGE("Wobbuffet used Scratch!");
+    }
+}
+
+WILD_BATTLE_TEST("Embargo doesn't block held item effects that affect friendship")
+{
+    u32 initialFriendship;
+    u32 finalFriendship;
+
+    KNOWN_FAILING; // Pokémon are currently not obtaining Friendship for using items in battle.
+    GIVEN {
+        ASSUME(gItems[ITEM_X_ACCURACY].battleUsage == EFFECT_ITEM_INCREASE_STAT);
+        PLAYER(SPECIES_WOBBUFFET) { Item(ITEM_SOOTHE_BELL); };
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { USE_ITEM(player, ITEM_X_ACCURACY); }
+        TURN { MOVE(player, MOVE_SING); }
+    } SCENE {
+        MESSAGE("Wobbuffet used Sing!");
+        MESSAGE("Wild Wobbuffet fell asleep!");
+    } THEN {
+        initialFriendship = GetMonData(&PLAYER_PARTY[0], MON_DATA_FRIENDSHIP);
+        finalFriendship = GetMonData(&gPlayerParty[0], MON_DATA_FRIENDSHIP);
+        EXPECT_EQ(finalFriendship, initialFriendship + 2);
+    }
+}
+
+SINGLE_BATTLE_TEST("Embargo doesn't block a held item's form-changing effect, but it does any other effects", s16 damage)
+{
+    u32 heldItem;
+
+    PARAMETRIZE { heldItem = ITEM_NONE; }
+    PARAMETRIZE { heldItem = ITEM_MEADOW_PLATE; }
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_ARCEUS_GRASS) { Item(heldItem); };
+        ASSUME(gItems[ITEM_MEADOW_PLATE].holdEffect == HOLD_EFFECT_PLATE);
+        ASSUME(gItems[ITEM_MEADOW_PLATE].holdEffectParam == 20);
+        ASSUME(gItems[ITEM_MEADOW_PLATE].secondaryId == TYPE_GRASS);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_RAZOR_LEAF); }
+    } SCENE {
+        HP_BAR(player, captureDamage: &results[i].damage);
+    } FINALLY {
+        EXPECT_MUL_EQ(results[0].damage, Q_4_12(1.2), results[1].damage);
+    }
+}
+
+SINGLE_BATTLE_TEST("Embargo makes Fling and Natural Gift fail")
+{
+    u32 heldItem, moveId;
+
+    PARAMETRIZE { heldItem = ITEM_LIGHT_BALL; moveId = MOVE_FLING; }
+    PARAMETRIZE { heldItem = ITEM_CHERI_BERRY; moveId = MOVE_NATURAL_GIFT; }
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Item(heldItem); };
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_EMBARGO); }
+        TURN { MOVE(player, moveId); }
+    } SCENE {
+        // Turn 1
+        MESSAGE("Foe Wobbuffet used Embargo!");
+        MESSAGE("Wobbuffet can't use items anymore!");
+        // Turn 2
+        if (moveId == MOVE_FLING)
+            MESSAGE("Wobbuffet used Fling!");
+        else
+            MESSAGE("Wobbuffet used Natural Gift!");
+        MESSAGE("But it failed!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Embargo doesn't stop an item flung at an affected target from activating")
```

Comment:
I don't get this test. What is exactly being checked here? Player uses Embargo on opponent, and then uses Fling.

Reply:
It tests that the effect of the item that is flung at a target that is suffering the effects of Embargo activates normally 😄

Reply:
In other words, it verifies that Embargo is only blocking the affected Pokémon's own item effects, and that they're not blocking item effects that would affect them.

I don't know if it's a silly test to make or not, I just wrote tests based on what's written at places such as Smogon and Bulbapedia 😅 ...

Reply:
Doesn't that mean we need a config for Gen IV behaviour then?

Reply:
This is what I was told on Discord when I was still writing the Embargo and Mega Evolution test:

> SBird (Karathan) 11/11/2023 8:56 AM
> the tests can't currently override preproc constants
> (I think)

:eyes: